notebook/notebookapp.py

```diff
@@ -1100,6 +1109,13 @@ def start(self):
         
         This method takes no arguments so all configuration and initialization
         must be done prior to calling this method."""
+        try:
+            if os.geteuid() == 0 and not self.allow_root:
+                self.log.critical("Running as root is forbidden. Use --allow-root to bypass.")
+                self.exit(1)
+        except OSError as e:
+            pass
```

Comment:
@takluyver Whoops, I read the [docs](https://docs.python.org/2/library/os.html) too fast 
> Note All functions in this module raise OSError in the case of invalid or inaccessible file names and paths, or other arguments that have the correct type, but are not accepted by the operating system.

Should be excepting AttributeError, my bad!